middleware.js

```diff
@@ -0,0 +1,68 @@
+import { NextResponse } from "next/server";
+import { Buffer } from "node:buffer";
+
+async function sendMetric(hostName, metricKey) {
+  await fetch(`${hostName}/api/metric`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json"
+    },
+    body: JSON.stringify({ metricKey })
+  });
+}
+
+export async function middleware(req) {
+  const cookie = req.headers.get("Cookie");
+  const split_url = req.url.split("/");
+  const HOST_NAME = split_url.slice(0, 3).join("/");
+  let _metricName;
+  if (req.url.includes("profile")) {
+    _metricName = split_url.slice(3, -1).join("_");
+  } else _metricName = split_url.slice(3).join("_");
+
+  // console.log(HOST_NAME, _metricName);
+
+  // skip the /api/metric api endpoint
+  if (req.url.includes("metric")) return NextResponse.json({ success: true });
+
+  try {
+    let response;
+    if (req.body) {
+      const rawBody = await req.body.getReader().read();
+      const body = JSON.parse(Buffer.from(rawBody?.value).toString("utf8"));
+      // const reqHeaders = Object.fromEntries(req.headers.entries());
+
+      response = await fetch(req.url, {
```

Comment:
What happens if this fetch takes 10 seconds to resolve.  Will the underlying request that kicked off this middleware API call block for 10s?